### DISCUSSION TITLE
How to add new custom cells when using the Extra Cells package?

### DISCUSSION THREAD

**Discussion by city17** [glideapps/glide-data-grid#411] [Q&A] [answered]
Hi, thanks for building this data grid, it's great to use and build with. One thing that's a bit harder to use is the adding of custom cells.

I've checked your source code for the dropdown cell and I think I can create a new custom cell based on that. But I don't know exactly how to add it to the data grid. Especially when already using the `glide-data-grid-cells` package (which already uses `provideEditor` and `drawCell`).

Could you give an example of how to add your own custom cell to the data grid, when using the extra cells package?

**Comment by city17** [comment, marked as answer]
Figured it out in the meantime, maybe this is useful for others:

Instead of using `provideEditor` and `drawCell` you can use `useCustomCells`, which you then spread on the DataEditor component.

`useCustomCells` takes an array  of custom cells, where you can add both your own and the ones from the `glide-data-grid-cells` package, like so:

```JSX
import { DataEditor, useCustomCells } from '@glideapps/glide-data-grid';
import { TagsCell } from '@glideapps/glide-data-grid-cells'; // Import each cell from this package individually now

import MyOwnCustomCell from './customCells/MyOwnCustomCell'

const customCells = useCustomCells([TagsCell, MyOwnCustomCell]);

<DataEditor {...customCells} />
```

**Comment by jassmith** [comment]
This is my favorite kind of thread to wake up to, self solving :)
